rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/CreateIssueSummaryDataRuleProvider.java

```diff
@@ -0,0 +1,132 @@
+package org.jboss.windup.rules.apps.java.reporting.rules;
+
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.metadata.MetadataBuilder;
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.config.phase.ReportRenderingPhase;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.WindupConfigurationModel;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.WindupConfigurationService;
+import org.jboss.windup.reporting.freemarker.problemsummary.ProblemSummary;
+import org.jboss.windup.reporting.freemarker.problemsummary.ProblemSummaryService;
+import org.jboss.windup.reporting.model.Severity;
+import org.jboss.windup.reporting.service.EffortReportService;
+import org.jboss.windup.reporting.service.ReportService;
+import org.jboss.windup.util.exception.WindupException;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.MappingJsonFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Generates a .js (javascript) file in the reports directory with detailed issue summary information.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class CreateIssueSummaryDataRuleProvider extends AbstractRuleProvider
```

Comment:
I meant to use freemarker to generate javascript instead of hardcoded java code that makes it hard to make a full picture of what is generated (issueSummaryWriter.write("WINDUP_ISSUE_SUMMARIES['" + inputApplication.asVertex().getId() + "'] = ");)